database/sql/src/main/scala/cromwell/database/slick/tables/CallCachingAggregationEntryComponent.scala

```diff
@@ -50,6 +50,25 @@ trait CallCachingAggregationEntryComponent {
     } yield ()).exists
   )
 
+  val existsCallCachingEntriesForBaseAggregationHashWithCallCachePrefix = Compiled(
+    (baseAggregation: Rep[String],
+     prefix1: Rep[String], prefix1Length: Rep[Int],
+     prefix2: Rep[String], prefix2Length: Rep[Int],
+     prefix3: Rep[String], prefix3Length: Rep[Int]
+    ) => (for {callCachingEntry <- callCachingEntries
+      callCachingEntry <- callCachingEntries
+      if callCachingEntry.allowResultReuse
+      callCachingAggregationEntry <- callCachingAggregationEntries
+      if callCachingEntry.callCachingEntryId === callCachingAggregationEntry.callCachingEntryId
+      if callCachingAggregationEntry.baseAggregation === baseAggregation
+      detritus <- callCachingDetritusEntries
+      if detritus.callCachingEntryId === callCachingEntry.callCachingEntryId
+      detritusPath = detritus.detritusValue.map { x => x.asColumnOf[String] }
+      if (detritusPath.substring(0, prefix1Length) === prefix1) ||
```

Comment:
ToL thinking you could do something with `Option[String]` args,  `flatten` a list of them, and filter by `startsWith` but not 100% sure it would be cleaner.  could get rid of the prefixlength and noMatch stuff.  Will checkout your PR and attempt

Reply:
I tried a few things before I ended up with this yukkiness and I think mostly due to the restrictions of `Compiled` ended up with this. At least once I had something that only blew up at runtime, so watch out for that. 🙂